pkg/grpc/server_test.go

```diff
@@ -1,33 +1,153 @@
 package grpc
 
 import (
-	"os"
+	"crypto/tls"
+	"fmt"
+	"net/http"
 	"testing"
+	"time"
 
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/grpc/authz/allow"
+	"github.com/stackrox/rox/pkg/grpc/routes"
+	"github.com/stackrox/rox/pkg/mtls/verifier"
+	"github.com/stretchr/testify/suite"
 )
 
-func TestMaxResponseMsgSize_Unset(t *testing.T) {
-	require.NoError(t, os.Unsetenv(maxResponseMsgSizeSetting.EnvVar()))
+type APIServerSuite struct {
+	suite.Suite
+}
+
+func (a *APIServerSuite) SetupTest() {
+	// In order to start the gRPC server, we need to have the MTLS environment variables
+	// pointing to some valid certificate/key pair. In this case we are using the ones
+	// created for local-sensor, which are dummy self-signed certificates.
+	a.T().Setenv("ROX_MTLS_CERT_FILE", "../../tools/local-sensor/certs/cert.pem")
+	a.T().Setenv("ROX_MTLS_KEY_FILE", "../../tools/local-sensor/certs/key.pem")
+	a.T().Setenv("ROX_MTLS_CA_FILE", "../../tools/local-sensor/certs/caCert.pem")
+	a.T().Setenv("ROX_MTLS_CA_KEY_FILE", "../../tools/local-sensor/certs/caKey.pem")
+}
+
+func Test_APIServerSuite(t *testing.T) {
```

Comment:
I think it would be good to add a test to verify the sequence start -> stop -> start works fine.

Reply:
Do you have an use case for `start -> stop -> start`? I can't think of why this should be done. `Stop()` should _only_ be used when stopping the application -- similar to what happened before, without the explicit `Stop()` call. 

As shown in the tests, what will happen when writing other tests, is that _different_ API instances can call `Start()`, and they should be able to register TCP listeners in the same ports.

 What I can offer is that `Start` should return an error if called after `Stop`. Would that work for you?

Reply:
> What I can offer is that Start should return an error if called after Stop. Would that work for you?

That works. If `start -> stop -> start` is not meant to happen, then I think signaling an error if `Start` is called twice makes sense. 

I think this phrase in the description confused me: 

> In order to close this technical debt and allow us to write tests starting and stopping the gRPC server multiple times

I thought the objective was to call `sensor.Start`, `sensor.Stop`, `sensor.Start`, etc. But what you said makes sense.

Following this, would it make sense to ensure `start -> start` also returns an error? I'm ok with leaving it as it is and letting it panic.